From 81dbf9730556c82f821d5f97d8838c5b505bd6e8 Mon Sep 17 00:00:00 2001
From: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
Date: Thu, 31 Aug 2017 11:52:03 +0530
Subject: [PATCH 135/155] armv8: ls1012a: Memory Map modification of kernel,env

This patch adjusts memory map for images on LS1012A
as per below memory map:
Image                               Flash Offset
RCW+PBI                             0x00000000
Boot firmware (U-Boot)              0x00100000
Boot firmware Environment           0x00300000

Signed-off-by: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
---
 include/configs/ls1012a_common.h |  4 ++--
 include/configs/ls1012afrdm.h    | 16 ++++++++--------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/include/configs/ls1012a_common.h b/include/configs/ls1012a_common.h
index 42bbc028d3..1f18b27d26 100644
--- a/include/configs/ls1012a_common.h
+++ b/include/configs/ls1012a_common.h
@@ -66,7 +66,7 @@
 
 #define CONFIG_ENV_IS_IN_SPI_FLASH
 #define CONFIG_ENV_SIZE			0x40000          /* 256KB */
-#define CONFIG_ENV_OFFSET		0x200000        /* 2MB */
+#define CONFIG_ENV_OFFSET		0x300000        /* 3MB */
 #define CONFIG_ENV_SECT_SIZE		0x40000
 #endif
 
@@ -98,7 +98,7 @@
 	"kernel_addr=0x100000\0"		\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"initrd_high=0xffffffffffffffff\0"	\
-	"kernel_start=0xa00000\0"		\
+	"kernel_start=0x1000000\0"		\
 	"kernel_load=0xa0000000\0"		\
 	"kernel_size=0x2800000\0"		\
 
diff --git a/include/configs/ls1012afrdm.h b/include/configs/ls1012afrdm.h
index 6b1ba578e9..8a9444d445 100644
--- a/include/configs/ls1012afrdm.h
+++ b/include/configs/ls1012afrdm.h
@@ -22,14 +22,14 @@
 
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS              \
-       "verify=no\0"                           \
-       "loadaddr=0x80100000\0"                 \
-       "kernel_addr=0x100000\0"                \
-       "fdt_high=0xffffffffffffffff\0"         \
-       "initrd_high=0xffffffffffffffff\0"      \
-       "kernel_start=0xa00000\0"               \
-       "kernel_load=0x96000000\0"              \
-       "kernel_size=0x2800000\0"
+	"verify=no\0"                           \
+	"loadaddr=0x80100000\0"                 \
+	"kernel_addr=0x100000\0"                \
+	"fdt_high=0xffffffffffffffff\0"         \
+	"initrd_high=0xffffffffffffffff\0"      \
+	"kernel_start=0x1000000\0"               \
+	"kernel_load=0x96000000\0"              \
+	"kernel_size=0x2800000\0"
 
 /*
 * USB
-- 
2.14.1

